src/blockstore.js

```diff
@@ -31,8 +30,27 @@ const cidToDsKey = (cid) => {
   return keyFromBuffer(cid.buffer)
 }
 
-module.exports = (repo) => {
-  const store = new NamespaceStore(repo.store, blockPrefix)
+module.exports = (filestore, options, callback) => {
+  maybeWithSharding(filestore, options, (err, store) => {
+    if (err) {
+      callback(err)
+      return // early
+    }
```

Comment:
I'm not against to `if (err) { return callback(err) }` if the goal is to save a couple of lines.